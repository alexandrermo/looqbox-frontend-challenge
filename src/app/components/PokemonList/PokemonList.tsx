'use client'

import Image from 'next/image'
import PokemonTypes from '@/types/PokemonTypes'
import { Card, Input } from 'antd'
import { useState } from 'react'
import Link from 'next/link'
import stringUtil from '@/utils/stringUtil/stringUtil'

interface Props {
    kantoPokedex: PokemonTypes.Pokedex
}

const PokemonList = (props: Props) => {
    const { kantoPokedex } = props

    const [searchText, setSearchText] = useState('')

    const pokemonsToShow = kantoPokedex.pokemon_entries.filter(
        (pokemonEntry) => {
            return (
                searchText === '' ||
                pokemonEntry.pokemon_species.name
                    .toLowerCase()
                    .includes(searchText.toLowerCase())
            )
        }
    )

    return (
        <>
            <Input.Search
                placeholder='Pesquisar'
                onSearch={(newSearchText) => {
                    setSearchText(newSearchText)
                }}
                style={{ maxWidth: 400 }}
            />

            <div className='grid grid-cols-12 w-full gap-4 mt-2'>
                {pokemonsToShow.map((pokemonEntry) => {
                    return (
                        <Card
                            key={pokemonEntry.entry_number}
                            className='col-span-12 sm:col-span-4 lg:col-span-3'
                            title={stringUtil.firstLetterUpper(
                                pokemonEntry.pokemon_species.name
                            )}
                            extra={
                                <Link
                                    href={`/pokemon/${pokemonEntry.pokemon_species.name}`}
                                >
                                    Details
                                </Link>
                            }
                        >
                            <div className='w-full h-28 relative'>
                                <Image
                                    alt={`Imagem de ${pokemonEntry.pokemon_species.name}`}
                                    className='object-contain'
                                    src={`https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/${pokemonEntry.entry_number}.png`}
                                    fill
                                />
                            </div>
                        </Card>
                    )
                })}
            </div>
        </>
    )
}

export default PokemonList
