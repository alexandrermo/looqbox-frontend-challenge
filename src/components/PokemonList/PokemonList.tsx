'use client'

import Image from 'next/image'
import PokemonTypes from '@/types/PokemonTypes'
import { Card, Input } from 'antd'
import { useState } from 'react'

interface Props {
    kantoPokedex: PokemonTypes.Pokedex
}

const PokemonList = (props: Props) => {
    const { kantoPokedex } = props

    const [searchText, setSearchText] = useState('')

    const pokemonsToShow = kantoPokedex.pokemon_entries.filter(
        (pokemonEntry) => {
            return (
                searchText === '' ||
                pokemonEntry.pokemon_species.name
                    .toLowerCase()
                    .includes(searchText.toLowerCase())
            )
        }
    )

    return (
        <>
            <Input.Search
                placeholder='Pesquisar'
                onSearch={(newSearchText) => {
                    setSearchText(newSearchText)
                }}
                style={{ maxWidth: 400 }}
            />

            <div className='grid grid-cols-12 w-full gap-4 mt-2'>
                {pokemonsToShow.map((pokemonEntry) => {
                    return (
                        <Card
                            key={pokemonEntry.entry_number}
                            className='col-span-12 sm:col-span-4 lg:col-span-3'
                            title={`${pokemonEntry.pokemon_species.name
                                .charAt(0)
                                .toUpperCase()}${pokemonEntry.pokemon_species.name.substring(
                                1
                            )}`}
                            extra={<a href='#'>More</a>}
                        >
                            <div className='w-full h-28 relative'>
                                <Image
                                    alt={`Imagem de ${pokemonEntry.pokemon_species.name}`}
                                    className='object-contain'
                                    src={`https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/${pokemonEntry.entry_number}.png`}
                                    fill
                                />
                            </div>
                        </Card>
                    )
                })}
            </div>
        </>
    )
}

export default PokemonList
