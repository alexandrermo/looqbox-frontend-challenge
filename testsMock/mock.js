/* eslint-disable max-lines */
export const mockKantoPokedex = {
    id: 2,
    name: 'kanto',
    is_main_series: true,
    descriptions: [
        {
            description: 'Rot/Blau/Gelb Kanto Dex',
            language: {
                name: 'de',
                url: 'https://pokeapi.co/api/v2/language/6/',
            },
        },
    ],
    names: [
        {
            name: 'Kanto',
            language: {
                name: 'de',
                url: 'https://pokeapi.co/api/v2/language/6/',
            },
        },
    ],
    pokemon_entries: [
        {
            entry_number: 1,
            pokemon_species: {
                name: 'bulbasaur',
                url: 'https://pokeapi.co/api/v2/pokemon-species/1/',
            },
        },
    ],
    region: {
        name: 'kanto',
        url: 'https://pokeapi.co/api/v2/region/1/',
    },
    version_groups: [
        {
            name: 'red-blue',
            url: 'https://pokeapi.co/api/v2/version-group/1/',
        },
    ],
}
