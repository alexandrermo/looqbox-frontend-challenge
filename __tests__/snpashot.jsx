import { render } from '@testing-library/react'

import { mockKantoPokedex } from '../testsMock/mock'
import PokemonList from '@/app/components/PokemonList/PokemonList'

it('renders homepage unchanged', () => {
    const { container } = render(
        <PokemonList kantoPokedex={mockKantoPokedex} />
    )
    expect(container).toMatchSnapshot()
})
