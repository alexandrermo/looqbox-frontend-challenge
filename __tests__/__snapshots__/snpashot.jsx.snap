// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders homepage unchanged 1`] = `
<div>
  <span
    class="ant-input-group-wrapper ant-input-group-wrapper-outlined css-dev-only-do-not-override-j9bb5n ant-input-search"
    style="max-width: 400px;"
  >
    <span
      class="ant-input-wrapper ant-input-group css-dev-only-do-not-override-j9bb5n"
    >
      <input
        class="ant-input css-dev-only-do-not-override-j9bb5n ant-input-outlined"
        placeholder="Pesquisar"
        type="text"
        value=""
      />
      <span
        class="ant-input-group-addon"
      >
        <button
          class="ant-btn css-dev-only-do-not-override-j9bb5n ant-btn-default ant-btn-icon-only ant-input-search-button"
          type="button"
        >
          <span
            class="ant-btn-icon"
          >
            <span
              aria-label="search"
              class="anticon anticon-search"
              role="img"
            >
              <svg
                aria-hidden="true"
                data-icon="search"
                fill="currentColor"
                focusable="false"
                height="1em"
                viewBox="64 64 896 896"
                width="1em"
              >
                <path
                  d="M909.6 854.5L649.9 594.8C690.2 542.7 712 479 712 412c0-80.2-31.3-155.4-87.9-212.1-56.6-56.7-132-87.9-212.1-87.9s-155.5 31.3-212.1 87.9C143.2 256.5 112 331.8 112 412c0 80.1 31.3 155.5 87.9 212.1C256.5 680.8 331.8 712 412 712c67 0 130.6-21.8 182.7-62l259.7 259.6a8.2 8.2 0 0011.6 0l43.6-43.5a8.2 8.2 0 000-11.6zM570.4 570.4C528 612.7 471.8 636 412 636s-116-23.3-158.4-65.6C211.3 528 188 471.8 188 412s23.3-116.1 65.6-158.4C296 211.3 352.2 188 412 188s116.1 23.2 158.4 65.6S636 352.2 636 412s-23.3 116.1-65.6 158.4z"
                />
              </svg>
            </span>
          </span>
        </button>
      </span>
    </span>
  </span>
  <div
    class="grid grid-cols-12 w-full gap-4 mt-2"
  >
    <div
      class="ant-card ant-card-bordered col-span-12 sm:col-span-4 lg:col-span-3 css-dev-only-do-not-override-j9bb5n"
    >
      <div
        class="ant-card-head"
      >
        <div
          class="ant-card-head-wrapper"
        >
          <div
            class="ant-card-head-title"
          >
            Bulbasaur
          </div>
          <div
            class="ant-card-extra"
          >
            <a
              href="/pokemon/bulbasaur"
            >
              Details
            </a>
          </div>
        </div>
      </div>
      <div
        class="ant-card-body"
      >
        <div
          class="w-full h-28 relative"
        >
          <img
            alt="Imagem de bulbasaur"
            class="object-contain"
            data-nimg="fill"
            decoding="async"
            loading="lazy"
            sizes="100vw"
            src="/_next/image?url=https%3A%2F%2Fraw.githubusercontent.com%2FPokeAPI%2Fsprites%2Fmaster%2Fsprites%2Fpokemon%2F1.png&w=3840&q=75"
            srcset="/_next/image?url=https%3A%2F%2Fraw.githubusercontent.com%2FPokeAPI%2Fsprites%2Fmaster%2Fsprites%2Fpokemon%2F1.png&w=640&q=75 640w, /_next/image?url=https%3A%2F%2Fraw.githubusercontent.com%2FPokeAPI%2Fsprites%2Fmaster%2Fsprites%2Fpokemon%2F1.png&w=750&q=75 750w, /_next/image?url=https%3A%2F%2Fraw.githubusercontent.com%2FPokeAPI%2Fsprites%2Fmaster%2Fsprites%2Fpokemon%2F1.png&w=828&q=75 828w, /_next/image?url=https%3A%2F%2Fraw.githubusercontent.com%2FPokeAPI%2Fsprites%2Fmaster%2Fsprites%2Fpokemon%2F1.png&w=1080&q=75 1080w, /_next/image?url=https%3A%2F%2Fraw.githubusercontent.com%2FPokeAPI%2Fsprites%2Fmaster%2Fsprites%2Fpokemon%2F1.png&w=1200&q=75 1200w, /_next/image?url=https%3A%2F%2Fraw.githubusercontent.com%2FPokeAPI%2Fsprites%2Fmaster%2Fsprites%2Fpokemon%2F1.png&w=1920&q=75 1920w, /_next/image?url=https%3A%2F%2Fraw.githubusercontent.com%2FPokeAPI%2Fsprites%2Fmaster%2Fsprites%2Fpokemon%2F1.png&w=2048&q=75 2048w, /_next/image?url=https%3A%2F%2Fraw.githubusercontent.com%2FPokeAPI%2Fsprites%2Fmaster%2Fsprites%2Fpokemon%2F1.png&w=3840&q=75 3840w"
            style="position: absolute; height: 100%; width: 100%; left: 0px; top: 0px; right: 0px; bottom: 0px; color: transparent;"
          />
        </div>
      </div>
    </div>
  </div>
</div>
`;
